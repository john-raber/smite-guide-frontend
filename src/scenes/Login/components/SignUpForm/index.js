import React, { Component } from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import { Form, Button } from "reactstrap";

import EmailInput from "../EmailInput/index";
import PasswordInput from "../PasswordInput/index";
import PasswordConfirmationInput from "../PasswordConfirmationInput/index";

import { createUser } from "../../services/currentUser/actions/index";

class SignUpForm extends Component {
  state = {
    email: "",
    password: "",
    passwordConfirmation: ""
  };

  handleFormSubmit = e => {
    const { password, passwordConfirmation } = this.state;
    const { createUser, history } = this.props;

    e.preventDefault();

    if (password !== passwordConfirmation) {
      alert("Passwords must match");
    }

    createUser({ user: this.state }).then(history.push("/profile"));
  };

  handleInputChange = (name, value) => {
    this.setState({
      [name]: value
    });
  };

  render() {
    return (
      <Form className="SignUpForm" onSubmit={this.handleFormSubmit}>
        <EmailInput
          value={this.state.email}
          onChange={this.handleInputChange}
        />
        <PasswordInput
          value={this.state.password}
          onChange={this.handleInputChange}
        />
        <PasswordConfirmationInput
          value={this.state.passwordConfirmation}
          onChange={this.handleInputChange}
        />
        <Button color="warning">Register</Button>
      </Form>
    );
  }
}

export default withRouter(
  connect(
    null,
    { createUser }
  )(SignUpForm)
);

export { SignUpForm };
