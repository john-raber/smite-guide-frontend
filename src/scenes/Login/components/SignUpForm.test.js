import React from "react";
import { shallow } from "enzyme";

import { SignUpForm } from "./SignUpForm/index";

describe("SignUpForm", () => {
  it("renders correctly", () => {
    let wrapper = shallow(<SignUpForm />);

    expect(wrapper.find(".SignUpForm").length).toEqual(1);
  });

  it("matches its snapshot", () => {
    let wrapper = shallow(<SignUpForm />);

    expect(wrapper).toMatchSnapshot();
  });
});
