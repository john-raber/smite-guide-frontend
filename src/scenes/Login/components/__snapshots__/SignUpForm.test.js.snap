// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm matches its snapshot 1`] = `
<Form
  className="SignUpForm"
  onSubmit={[Function]}
  tag="form"
>
  <EmailInput
    onChange={[Function]}
    value=""
  />
  <PasswordInput
    onChange={[Function]}
    value=""
  />
  <PasswordConfirmationInput
    onChange={[Function]}
    value=""
  />
  <Button
    color="warning"
    tag="button"
  >
    Register
  </Button>
</Form>
`;
